base/compiler/abstractinterpretation.jl

```diff
@@ -2226,6 +2226,18 @@ function abstract_throw(interp::AbstractInterpreter, argtypes::Vector{Any}, ::Ab
     return CallMeta(Union{}, exct, EFFECTS_THROWS, NoCallInfo())
 end
 
+function abstract_throw_methoderror(interp::AbstractInterpreter, argtypes::Vector{Any}, ::AbsIntState)
+    na = length(argtypes)
+    exct = if na == 1
+        ArgumentError
+    elseif !isvarargtype(argtypes[2])
+        MethodError
+    else
+        tmerge(𝕃ᵢ, MethodError, ArgumentError)
+    end
+    return CallMeta(Union{}, exct, EFFECTS_THROWS, NoCallInfo())
```

Comment:
I guess its not consistent because it captures the world?

Reply:
Great point - I got that right in the tfuncs but not here.

Reply:
Actually, I think this is consistent after all.

Consistency doesn't place requirements on exceptions:
```
   If :consistent functions terminate by throwing an exception, that exception
    itself is not required to meet the egality requirement specified above.
```

And even if it did, it is per-world-age:
```
 The :consistent-cy assertion is made world-age wise. More formally, write
    fᵢ for the evaluation of f in world-age i, then this setting requires:
    ∀ i, x, y: x ≡ y → fᵢ(x) ≡ fᵢ(y)
    However, for two world ages i, j s.t. i ≠ j, we may have fᵢ(x) ≢ fⱼ(y).
```

Reply:
Yeah, `:consistent` doesn't require the consistency on thrown object. If a method throws consistently, then it is `:consistent` no matter what it throws.

Reply:
wait that is a kind of odd spec for it to have. Does that mean we can't inline the throw of an exception if we think a function throws?

Reply:
Yeah, even if concrete evaluation reveals that a method call will definitely throw, we can't propagate that exception as a `Const`.